### ISSUE TITLE
Reverse proxy sends mobile Safari into refresh loop

### ISSUE THREAD

**Issue by Densaugeo** [caddyserver/caddy#5360]
Steps to reproduce:

1 - Run a simple HTTP server `python -m http.server 8000`
2 - Start caddy reverse proxy `caddy reverse-proxy --from https://your_pc.local:443 --to http://localhost:8000`
3 - Open `https://your_pc.local/` on another device (in a browser other than mobile Safari). http.server's directory listing loads normally.
4 - Open `https://your_pc.local/` in mobile Safari. Instead of loading, page gets stuck continually refreshing. The address bar shows flickering text that changes too quickly for me to read.

I tested this proxy setup with Firefox on Linux, Firefox on Windows, Chrome on Windows, and Firefox on Android. The proxy setup works will all of those. It only fails on mobile Safari.

Caddy version: tested with 2.6.2 (latest) and 2.3.0 (Fedora repo).
Mobile Safari/iOS version: 16.1.

**Comment by francislavoie** [comment]
Try using a different domain than `.local`. That TLD is used by Apple devices for multicast DNS, and it can really mess things up with routing.

I don't own any Apple devices, so I can't test this. You'll need to provide more specific details about the behaviour. Enable the `debug` global option, and show your logs. Try to capture the headers from a request on your mobile device (I'm sure there are iOS apps that could help with debugging).

Either way, I highly doubt this is a bug with Caddy, but instead either a misconfiguration or unfortunate network circumstances.

Please ask your usage questions on the [Caddy community forums](https://caddy.community). We prefer to keep the GitHub issue board for bugs and feature requests. Don't forget to fill out the thread template so we can help you!

**Comment by Densaugeo** [comment]
Thank you for the advice. I retested without `.local`, but it made no difference. To test if the issue is related to Caddy, I tried serving HTTPS directly from another Python server, and the iPhone was able to open it without issue. The problem only occurs when proxied through Caddy.

For the new test, I ran `caddy run` with the following Caddyfile:

~~~
{
  admin off
  debug
}

https://morpheus:443 {
  tls internal
  reverse_proxy http://localhost:8000
}
~~~

I also tested with the proxy listening at :8000 and the python server at :9000, but this did not have any effect either.

Adding the debug setting shows some sort of TLS handshake error is occurring:

~~~
2023/02/06 06:59:06.770 DEBUG   events  event   {"name": "tls_get_certificate", "id": "b81aab03-56c7-4359-9a70-e5a0118146cf", "origin": "tls", "data": {"client_hello":{"CipherSuites":[31354,4865,4866,4867,49196,49195,52393,49200,49199,52392,49162,49161,49172,49171,157,156,53,47,49160,49170,10],"ServerName":"morpheus","SupportedCurves":[19018,29,23,24,25],"SupportedPoints":"AA==","SignatureSchemes":[1027,2052,1025,1283,515,2053,2053,1281,2054,1537,513],"SupportedProtos":["h2","http/1.1"],"SupportedVersions":[56026,772,771,770,769],"Conn":{}}}}
2023/02/06 06:59:06.770 DEBUG   tls.handshake   choosing certificate    {"identifier": "morpheus", "num_choices": 1}
2023/02/06 06:59:06.770 DEBUG   tls.handshake   default certificate selection results   {"identifier": "morpheus", "subjects": ["morpheus"], "managed": true, "issuer_key": "local", "hash": "9fc8b88339d463ffd9bbfd4134747f78d40f025e76adb58a3b78c37abd165820"}
2023/02/06 06:59:06.770 DEBUG   tls.handshake   matched certificate in cache    {"remote_ip": "192.168.11.141", "remote_port": "64361", "subjects": ["morpheus"], "managed": true, "expiration": "2023/02/06 18:36:38.000", "hash": "9fc8b88339d463ffd9bbfd4134747f78d40f025e76adb58a3b78c37abd165820"}
2023/02/06 06:59:06.790 DEBUG   http.stdlib     http: TLS handshake error from 192.168.11.141:64361: EOF
2023/02/06 06:59:06.802 DEBUG   events  event   {"name": "tls_get_certificate", "id": "01fd612d-deab-4f54-a1d0-6d85d36a83d6", "origin": "tls", "data": {"client_hello":{"CipherSuites":[27242,4865,4866,4867,49196,49195,52393,49200,49199,52392,49162,49161,49172,49171,157,156,53,47,49160,49170,10],"ServerName":"morpheus","SupportedCurves":[39578,29,23,24,25],"SupportedPoints":"AA==","SignatureSchemes":[1027,2052,1025,1283,515,2053,2053,1281,2054,1537,513],"SupportedProtos":["h2","http/1.1"],"SupportedVersions":[35466,772,771,770,769],"Conn":{}}}}
2023/02/06 06:59:06.802 DEBUG   tls.handshake   choosing certificate    {"identifier": "morpheus", "num_choices": 1}
2023/02/06 06:59:06.802 DEBUG   tls.handshake   default certificate selection results   {"identifier": "morpheus", "subjects": ["morpheus"], "managed": true, "issuer_key": "local", "hash": "9fc8b88339d463ffd9bbfd4134747f78d40f025e76adb58a3b78c37abd165820"}
2023/02/06 06:59:06.802 DEBUG   tls.handshake   matched certificate in cache    {"remote_ip": "192.168.11.141", "remote_port": "64362", "subjects": ["morpheus"], "managed": true, "expiration": "2023/02/06 18:36:38.000", "hash": "9fc8b88339d463ffd9bbfd4134747f78d40f025e76adb58a3b78c37abd165820"}
2023/02/06 06:59:06.826 DEBUG   http.stdlib     http: TLS handshake error from 192.168.11.141:64362: EOF
~~~

(It repeats in a similar way every few ms).
